stdlib/public/core/Span/RawSpan.swift

```diff
@@ -97,9 +97,11 @@ extension RawSpan {
   public init(
     _unsafeBytes buffer: UnsafeRawBufferPointer
   ) {
-    self.init(
-      _unchecked: buffer.baseAddress, byteCount: buffer.count
-    )
+    let baseAddress = buffer.baseAddress
```

Comment:
Is there a reason a local variable needs to be created for this trivial expression?

Reply:
The `baseAddress` variable is the source of a lifetime dependence. Assigning it to a variable gives it a lexical scope. The lifetime of an expression ends after evaluation of the expression.

Reply:
I updated the commit log and PR comments.

Reply:
Does this mean that if you created `span` by calling `RawSpan(_unchecked: buffer.baseAddress, byteCount: buffer.count)`, that `span`'s lifetime would be limited to essentially just that initialization statement?

Reply:
Yes, the Span initializer says that it "borrows" the base address. If the base address is a sub-expression then the initializer borrows the temporary result of the sub-expression, which is only valid in the expression in which it is used. Here, baseAddress is a computed property, so the compiler just sees an expression. It doesn't know that baseAddress actually is a member of buffer.